src/Controller/Backend/PageRevisionBackendController.php

```diff
@@ -206,20 +209,26 @@ public function publishAction(
             $pageRevision->setPublished(true);
             $pageRevision->setUpdatedAt(new \DateTime());
 
-            $this->removePageContent($pageRevision);
+            $this->purgePageContent($pageRevision);
 
             $this->pageRevisionRepository->save($pageRevision, true);
 
+            $isDynamic = $this->hasDynamicShortcode($pageRevision, $shortcodeManager);
+
+            $pageRawQuery->update($pageRevision->getPage()->getId(), [
+                'dynamic' => $isDynamic,
+            ]);
```

Comment:
Using the raw query service here to avoid the Page entity from being saved and triggering listener events that don't need to happen.